components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt

```diff
@@ -133,21 +133,21 @@ class FirefoxAccount internal constructor(
         }
     }
 
-    override fun authorizeOAuthCodeAsync(
+    override suspend fun authorizeOAuthCode(
         clientId: String,
         scopes: Array<String>,
         state: String,
         accessType: AccessType
-    ) = scope.async {
+    ) = withContext(scope.coroutineContext) {
         handleFxaExceptions(logger, "authorizeOAuthCode", { null }) {
             inner.authorizeOAuthCode(clientId, scopes, state, accessType.msg)
         }
     }
 
     override fun getSessionToken(): String? {
```

Comment:
Do any of your consumers actually use `getSessionToken`, or is it exposed mostly for completeness? It doesn't feel like anyone should be calling this method in practice.

Reply:
In practice, this is just used internally by the account manager (to get sync going).

In theory, if consumers aren't using the account manager this lets them do something useful with the account object beyond just fetching a profile, but... yeah, not really something we want to encourage.

Next on my plate is to work through the "we have a firefox-shaped oauth account abstraction with just a single implementation" bit, and simplify things on that front... Perhaps we can clean this up as part of that work.